stackslib/src/net/httpcore.rs

```diff
@@ -1903,15 +1907,24 @@ pub fn send_http_request(
         if Instant::now().saturating_duration_since(start) > timeout {
             return Err(io::Error::new(
                 io::ErrorKind::WouldBlock,
-                "Timed out while receiving request",
+                "Timed out while sending request",
             ));
         }
+        // Heartbeat log
+        if Instant::now().saturating_duration_since(last_heartbeat_time) >= HEARTBEAT_INTERVAL {
```

Comment:
I think using `last_heartbeat_time.elapsed()` is cleaner. Maybe you can update the existing earlier check too?

Reply:
I think changing all of these to use `elapsed` is best.